src/libraries/System.Text.Json/src/System/Text/Json/Writer/JsonWriterHelper.cs

```diff
@@ -10,28 +10,40 @@ namespace System.Text.Json
 {
     internal static partial class JsonWriterHelper
     {
-        public static void WriteIndentation(Span<byte> buffer, int indent)
+        public static void WriteIndentation(Span<byte> buffer, int indent, byte indentByte)
         {
-            Debug.Assert(indent % JsonConstants.SpacesPerIndent == 0);
             Debug.Assert(buffer.Length >= indent);
 
             // Based on perf tests, the break-even point where vectorized Fill is faster
             // than explicitly writing the space in a loop is 8.
-            if (indent < 8)
+            if (indent is < 8)
             {
                 int i = 0;
                 while (i < indent)
                 {
-                    buffer[i++] = JsonConstants.Space;
-                    buffer[i++] = JsonConstants.Space;
+                    buffer[i++] = indentByte;
                 }
             }
-            else
+            else if (indent is not 0)
```

Comment:
if this is meant to check that `indent` isn't zero, wouldn't that case fall into the previous branch in any case?

Reply:
Yes, you are right. Also removed.